### DISCUSSION TITLE
I want to improve accuracy in ch11.2.3.

### DISCUSSION THREAD

**Discussion by AyanoSagasaki** [rasbt/machine-learning-book#153] [Q&A]
I tried to code the neural network in ch11. But when I experiment using MNIST, I can't improve accuracy.
I use same code, and same parameter in [this page](https://github.com/rasbt/machine-learning-book/blob/main/ch11/ch11.ipynb). But not same accuracy in the page.
![image](https://github.com/rasbt/machine-learning-book/assets/114718837/a903366a-4f7e-485c-af12-84fa331468dc)

**Comment by rasbt** [comment]
Hi @AyanoSagasaki I just reran the code on my computer and Google Colab and don't have this issue. Could you try on Google Colab? If you use the exact code from this repository and can't reproduce the results, it's maybe due to some older software libraries as my best guess.

**Reply by AyanoSagasaki** [reply]
Thank you for your advice, but I can't improve when I tried on Google Colab, too.
Incidentally, I used JupyterLab when I posted first discussion.
And libraries' version is, 
> numpy == 1.21.2
> scikit-learn == 1.0

![image](https://github.com/rasbt/machine-learning-book/assets/114718837/7daedeef-38ab-4af2-a22d-5b2abe4e8c47)

**Reply by rasbt** [reply]
That's so weird. I just ran it again with the default package versions in Google Colab and it seems to work just fine. You can access the code here: https://colab.research.google.com/drive/1LRwzT2R9nlNmLXvmakE8NVfkEc1P73q5?usp=sharing if you want to try.

You can either disable the

    from python_environment_check import check_packages

cell, or drag and drop the `python_environment_check.py` file into the Colab UI
